enumerable.rb

```diff
@@ -1,87 +1,146 @@
+# frozen_string_literal: true
+
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+# module Enumerable
 module Enumerable
   def my_each
-    for i in self do
-        self.class == Array ? yield(i) : yield(i[0], i[1])
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      instance_of?(Hash) ? yield([to_a[i][0], to_a[i][1]]) : yield(to_a[i])
+      i += 1
     end
+    self
   end
 
   def my_each_with_index
-    for i in self do
-      self.class == Array ? yield(i, self.index(i)) : yield([i[0], i[1]], self.keys.index(i[0]))
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      instance_of?(Hash) ? yield([to_a[i][0], to_a[i][1]], i) : yield(to_a[i], i)
+      i += 1
     end
+    self
   end
 
   def my_select
```

Comment:
- [ ] my_select returns an **array** containing all elements of enum for which the given block returns a true value::range. 

```
  range = Range.new(5,50)
  block = proc { |num| num < (0 + 9) / 2 }
  p range.my_select(&block) // returns { }
  p range.select(&block) // returns [ ]
```